base/array.jl

```diff
@@ -2363,11 +2363,14 @@ end
 ## Filter ##
 
 """
-    filter(f, a::AbstractArray)
+    filter(f, a)
```

Comment:
```suggestion
    filter(f, itr)
```

Reply:
> Does this support arbitrary iterables?

No, `Base.filter` is implemented only for specific types. So `itr` migth give the wrong idea here. But I would support having `filter` extended to arbitrary iterables, returning an `Vector` in this case, like `map` does.

Reply:
Ignore my suggestion then. If it changes to general iterables, we can update.